Code/GraphMol/catch_chirality.cpp

```diff
@@ -0,0 +1,829 @@
+    for (const auto &si : stereoInfo) {
+      CHECK(si.type == Chirality::StereoType::Atom_Tetrahedral);
+      CHECK(si.centeredOn % 2 == 0);
+      CHECK(si.specified == Chirality::StereoSpecified::Unspecified);
+    }
+  }
+}
+
+TEST_CASE("ring stereochemistry", "[chirality]") {
+  SECTION("specified") {
+    auto mol = "C[C@H]1CC[C@@H](C)CC1"_smiles;
+    REQUIRE(mol);
+    // std::cerr << "------------ 1 -------------" << std::endl;
+
+    auto stereoInfo = Chirality::findPotentialStereo(*mol);
+    REQUIRE(stereoInfo.size() == 2);
+    CHECK(stereoInfo[0].type == Chirality::StereoType::Atom_Tetrahedral);
+    CHECK(stereoInfo[0].centeredOn == 1);
+    CHECK(stereoInfo[0].specified == Chirality::StereoSpecified::Specified);
+    CHECK(stereoInfo[1].type == Chirality::StereoType::Atom_Tetrahedral);
+    CHECK(stereoInfo[1].centeredOn == 4);
+    CHECK(stereoInfo[1].specified == Chirality::StereoSpecified::Specified);
+  }
+  SECTION("unspecified") {
+    auto mol = "CC1CCC(C)CC1"_smiles;
+    REQUIRE(mol);
+    // std::cerr << "------------ 2 -------------" << std::endl;
+    auto stereoInfo = Chirality::findPotentialStereo(*mol);
+    REQUIRE(stereoInfo.size() == 2);
+    CHECK(stereoInfo[0].type == Chirality::StereoType::Atom_Tetrahedral);
+    CHECK(stereoInfo[0].centeredOn == 1);
+    CHECK(stereoInfo[0].specified == Chirality::StereoSpecified::Unspecified);
+
+    CHECK(stereoInfo[1].type == Chirality::StereoType::Atom_Tetrahedral);
+    CHECK(stereoInfo[1].centeredOn == 4);
+    CHECK(stereoInfo[1].specified == Chirality::StereoSpecified::Unspecified);
+  }
+  SECTION("four ring") {
+    auto mol = "C[C@H]1C[C@@H](C)C1"_smiles;
+    auto stereoInfo = Chirality::findPotentialStereo(*mol);
+    REQUIRE(stereoInfo.size() == 2);
+    CHECK(stereoInfo[0].type == Chirality::StereoType::Atom_Tetrahedral);
+    CHECK(stereoInfo[0].centeredOn == 1);
+    CHECK(stereoInfo[0].specified == Chirality::StereoSpecified::Specified);
+
+    CHECK(stereoInfo[1].type == Chirality::StereoType::Atom_Tetrahedral);
+    CHECK(stereoInfo[1].centeredOn == 3);
+    CHECK(stereoInfo[1].specified == Chirality::StereoSpecified::Specified);
+  }
+  SECTION("four ring unspecified") {
+    auto mol = "CC1CC(C)C1"_smiles;
+    auto stereoInfo = Chirality::findPotentialStereo(*mol);
+    REQUIRE(stereoInfo.size() == 2);
+    CHECK(stereoInfo[0].type == Chirality::StereoType::Atom_Tetrahedral);
+    CHECK(stereoInfo[0].centeredOn == 1);
+    CHECK(stereoInfo[0].specified == Chirality::StereoSpecified::Unspecified);
+
+    CHECK(stereoInfo[1].type == Chirality::StereoType::Atom_Tetrahedral);
+    CHECK(stereoInfo[1].centeredOn == 3);
+    CHECK(stereoInfo[1].specified == Chirality::StereoSpecified::Unspecified);
+  }
+}
+#if 0
+// FIX: the double bond stereo in rings isn't working. This also fails with the canonicalizer, so it's not unique to this code
+TEST_CASE("tricky recursive example from Dan Nealschneider", "[chirality]") {
+  SECTION("adapted") {
```

Comment:
I've been looking for something like https://docs.pytest.org/en/latest/skipping.html#xfail-mark-test-functions-as-expected-to-fail in catch, and haven't found it. bummer.

It feels like it would be even _more_ useful in C++, where you'd want the compiler to keep checking your tests.